src/provider.tsx

```diff
@@ -21,7 +21,7 @@ export const Provider = function<S extends Store>(props: PropsWithChildren<Props
   
   const [initialized, setInitialized] = useState(false)
   function onChange(value: StoreV<S>) {
-    setInitialized(true)
+    !initialized && setInitialized(true)
```

Comment:
If `initialized` is `true`, calling `setInitialized(true)` will not trigger re-rendering. React will automatically ignore it.

Reply:
yes, state is not change, so will not be-rerender, but the provider func will invoked.
i will show you a picture via `console.log`

Reply:
so i print the `provider render` str here:
![MCWQAO.png](https://s2.ax1x.com/2019/11/06/MCWQAO.png)

if `!initialized && setInitialized(true)`:
![MCW2D0.png](https://s2.ax1x.com/2019/11/06/MCW2D0.png)

if `setInitialized(true)`:
![MCWT29.png](https://s2.ax1x.com/2019/11/06/MCWT29.png)

Am I thinking wrong?

Reply:
Oh! I think I need to have a check.

Reply:
@awmleer 
I am learning React. This state management tool is my favorite at the moment, but I encountered a problem when debugging the execution process through debug. I thought for a long time and didn't find the reason.
as the demo code:
```jsx
function changeStore() {
    fooStore.setCounter(fooStore.counter + 1)
}
```
when i clicked the button change, next will execute this:
```js
    props.onChange((props.args ? props.useStore(...props.args) : props.useStore()));
```
so `Executor` rerender to execute the onChange callback, but why the `Executor` rerender?
I don't think it is a consumer, nor a change in status. I am very confused

I am looking forward to receiving a reply, thank you

Reply:
That's because `fooStore.setCounter` is actually the return value of `useState` in `Executor` component. So when you call `fooStore.setCounter(fooStore.counter + 1)`, you actually trigger the state update of `Executor`.